presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -568,4 +580,42 @@ public QueryContext getQueryContext(QueryId queryId)
     {
         return queryContexts.getUnchecked(queryId);
     }
+
+    @Override
+    public BooleanIntImmutablePair shouldBackPressure(TaskUpdateRequest taskUpdateRequest)
```

Comment:
Instead of using `BooleanIntImmutablePair`, which is difficult to parse, could you simply return a sentinel value for the retry after?  e.g. 0?